File: examples/src/Shapes/DuotoneEffectsPVI.cs
using System
import aspose.pydrawing as drawing
using Aspose.slides.Export
import aspose.slides as slides
using Aspose.slides.Effects

/*
This code demonstrates an operation where we added a picture for a slide background, added Duotone effect with styled colors, 
and then we got the effective duotone colors with which the background will be rendered.
*/

namespace Aspose.slides.Examples.CSharp.shapes
{
    class DuotoneEffectsPVI
    {
        public static void Run()
        {
            with slides.Presentation() as presentation:
            {
                imagePath = RunExamples.GetDataDir_Shapes() + "aspose-logo.jpg"

                # Add image to presentation
                backgroundImage = presentation.images.add_image(Image.FromFile(imagePath))

                # Set background in first slide
                presentation.slides[0].Background.type = BackgroundType.OwnBackground
                presentation.slides[0].Background.FillFormat.fill_type = slides.FillType.PICTURE
                presentation.slides[0].Background.FillFormat.picture_fill_format.picture.image = backgroundImage

                # Add Duotone effect to background
                IDuotone duotone = presentation.slides[0].Background.FillFormat.picture_fill_format.picture.ImageTransform
                    .AddDuotoneEffect()

                # Set Doutone properties
                duotone.Color1.ColorType = ColorType.Scheme
                duotone.Color1.SchemeColor = SchemeColor.Accent1
                duotone.Color2.ColorType = ColorType.Scheme
                duotone.Color2.SchemeColor = SchemeColor.Dark2

                # Get Effective values of the Duotone effect
                IDuotoneEffectiveData duotoneEffective = duotone.GetEffective()

                # Show effective values
                print("Duotone effective color1: " + duotoneEffective.Color1)
                print("Duotone effective color2: " + duotoneEffective.Color2)
            }
        }
    }
}

File: examples/src/Shapes/DuotoneEffectsPVI.py
import aspose.pydrawing as drawing
import aspose.slides as slides

"""
This code demonstrates an operation where we added a picture for a slide background, added Duotone effect with styled colors, 
and then we got the effective duotone colors with which the background will be rendered.
"""
dataDir = "./examples/data/"
outDir = "./examples/out/"

with slides.Presentation() as presentation:
    # Add image to presentation
    backgroundImage = presentation.images.add_image(drawing.Image.from_file(dataDir + "image1.jpg"))

    # Set background in first slide
    presentation.slides[0].background.type = slides.BackgroundType.OWN_BACKGROUND
    presentation.slides[0].background.fill_format.fill_type = slides.FillType.PICTURE
    presentation.slides[0].background.fill_format.picture_fill_format.picture.image = backgroundImage

    # Add Duotone effect to background
    duotone = presentation.slides[0].background.fill_format.picture_fill_format.picture.image_transform.add_duotone_effect()

    # Set Doutone properties
    duotone.color1.color_type = slides.ColorType.SCHEME
    duotone.color1.scheme_color = slides.SchemeColor.ACCENT1
    duotone.color2.color_type = slides.ColorType.SCHEME
    duotone.color2.scheme_color = slides.SchemeColor.DARK2

    # Get Effective values of the Duotone effect
    duotoneEffective = duotone.get_effective()

    # Show effective values
    print("Duotone effective color1: " + str(duotoneEffective.color1))
    print("Duotone effective color2: " + str(duotoneEffective.color2))
